packages/http-client-csharp/generator/TestProjects/CadlRanch.Tests/Http/_Type/Model/Inheritance/SingleDiscriminator/EagleTests.cs

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ClientModel;
+using System.IO;
+using _Type.Model.Inheritance.SingleDiscriminator.Models;
+using Microsoft.Generator.CSharp.Tests.Common;
+using NUnit.Framework;
+using TestProjects.CadlRanch.Tests.Infrastructure;
+
+namespace TestProjects.CadlRanch.Tests.Http._Type.Model.Inheritance.SingleDiscriminator
+{
+    internal class EagleTests : CadlRanchModelJsonTests<Eagle>
+    {
+        protected override string JsonPayload => File.ReadAllText(ModelTestHelper.GetLocation("TestData/Eagle/Eagle.json"));
+
+        protected override string WirePayload => File.ReadAllText(ModelTestHelper.GetLocation("TestData/Eagle/EagleWire.json"));
+
+        protected override void CompareModels(Eagle model, Eagle model2, string format)
+        {
+            Assert.AreEqual(model.Wingspan, model2.Wingspan);
+            Assert.AreEqual(model.Partner.Wingspan, model2.Partner.Wingspan);
+            Assert.AreEqual(model.Partner.GetType(), model2.Partner.GetType());
+            Assert.AreEqual(model.Friends.Count, model2.Friends.Count);
+            Assert.AreEqual(model.Friends[0].Wingspan, model2.Friends[0].Wingspan);
+            Assert.AreEqual(model.Friends[0].GetType(), model2.Friends[0].GetType());
+            Assert.AreEqual(model.Hate.Count, model2.Hate.Count);
+            Assert.AreEqual(model.Hate["key3"].Wingspan, model2.Hate["key3"].Wingspan);
+            Assert.AreEqual(model.Hate["key3"].GetType(), model2.Hate["key3"].GetType());
+        }
+
+        protected override BinaryContent ToBinaryContent(Eagle model) => model;
+
+        protected override Eagle ToModel(ClientResult result) => (Eagle)result;
+
+        protected override void VerifyModel(Eagle model, string format)
```

Comment:
Not familiar with this mechanism, if we have multiple tests inherit `CadlRanchModelJsonTests`, will there be multiple test runs for them?
And if one of them fails, will the test result show the specific failing for the one failed?

Reply:
yeah each model will have its own instance of the test and if one of 10 fails only the model that failed will show.

Reply:
We use the same thing in unbranded-typespec library you can see al models have their own 19 tests
![image](https://github.com/user-attachments/assets/b3a003a5-ef50-4f6c-a48e-14c7018d9b23)